featL added edit profile,follow/unfollow features

Chakravarthi0 opened the pull request:
now users can:
 - change their profile picture
 - change bio and url
 - follow/unfollow other users

Chakravarthi0 committed:
moved the postModal state from context to redux

Chakravarthi0 committed:
feat: added close modal on outside click hook

Chakravarthi0 committed:
backend related changes

Chakravarthi0 committed:
updated show side bar logic

Chakravarthi0 committed:
removed show post modal context provider

Chakravarthi0 committed:
chore: made profile page dynamic

Chakravarthi0 committed:
feat: added userSlices

Chakravarthi0 committed:
feat: Added get user slice

Chakravarthi0 committed:
feat: added follow/unfollow slice

Chakravarthi0 committed:
removed show post context

Chakravarthi0 committed:
feat: added user and follow slices to store

Chakravarthi0 committed:
feat:  added userlistmodal

Chakravarthi0 committed:
feat: implemented edit profile, follow/unfollow

pritam-kr commented on src/features/user/helpers/followUser.js:
```diff
@@ -0,0 +1,24 @@
+      toast.success("Followed user!");
+      return data;
+    } catch (err) {
+      console.log(err);
```
remove the consoles

pritam-kr reviewed (approved):
You have written good clean code.
good folder sructure.
 suggestion: 
 - the width of the post button on the feed page can be increase ??
 - toggle theme button looking more CTA to the user .

Other than everything is good to merge.

gopalbbharadva commented on src/components/edit-profile-modal/EditProfileModal.jsx:
```diff
@@ -1,40 +1,109 @@
               />
-              <div className="absolute right-0 bottom-0 flex justify-center items-center h-9 w-9 bg-black  text-white dark:bg-white dark:text-black rounded-full">
+              {/* <div className="absolute right-0 bottom-0 flex justify-center items-center h-9 w-9 bg-black  text-white dark:bg-white dark:text-black rounded-full">
                 <FiCamera className="text-xl cursor-pointer" />
-              </div>
+              </div> */}
```
You can remove these comments if not needed.

gopalbbharadva commented on src/components/post/Post.jsx:
```diff
@@ -1,42 +1,48 @@
+  const optionsModalRef = useRef(null);
+  const dispatch = useDispatch();
   const navigate = useNavigate();
   const toggleShowOptions = () => {
     setShowPostOptions((prev) => !prev);
   };
```
it can write in single line.

gopalbbharadva commented on src/features/user/helpers/getUsers.js:
```diff
@@ -0,0 +1,14 @@
+    const { data } = await axios.get("/api/users");
+    return data;
+  } catch (err) {
+    console.log(err);
```
you can remove this console

gopalbbharadva reviewed (approved):
Feature working fine, edit use details also working fine.
Bug: Updated user profile image in not sowing in posts avatar of that user.

manojasarsa reviewed (approved):
The code is pretty well structured.
Also, the functionality is working as expected.
Good work!

msdeshmukh009 reviewed (approved):
**Feature is working as expected.**

**Suggestions**: 
- You can close the followers/following modal after the user gets redirected to any user's profile. 
- I think the avatar on the profile page has a fixed height, it's better to use relative units there otherwise it will overflow.

Chakravarthi0 committed:
feat: added default profile picture

Chakravarthi0 committed:
style: changed modal bg color

Chakravarthi0 committed:
feat: added profile image uploading state

Chakravarthi0 committed:
chore: moved theme from context to redux store

Chakravarthi0 committed:
fix: fixed following/followers modal bug

Chakravarthi0 commented:
> **Feature is working as expected.**
> 
> **Suggestions**:
> 
> * You can close the followers/following modal after the user gets redirected to any user's profile.
> * I think the avatar on the profile page has a fixed height, it's better to use relative units there otherwise it will overflow.

fixed, thanks for reviewing

Chakravarthi0 commented:
> Feature working fine, edit use details also working fine. Bug: Updated user profile image in not sowing in posts avatar of that user.

It takes a while for the image to get uploaded, so, for a bried moment, the profile image will be empty. Added a loader to indicate that image is being uploaded, thanks for reviewing

Chakravarthi0 commented:
> You have written good clean code. good folder sructure. suggestion:
> 
> * the width of the post button on the feed page can be increase ??
> * toggle theme button looking more CTA to the user .
> 
> Other than everything is good to merge.

fixed, thanks for reviewing
